src/context/__tests__/auth.test.tsx

```
@@ -0,0 +1,753 @@
+      await act(async () => {
+        syncDataPromise = result.current.syncData();
+      });
+
+      // Check that syncing status is true while sync is in progress
+      await waitFor(() => {
+        expect(result.current.syncStatus.syncing).toBe(true);
+      });
+
+      // Resolve the sync
+      await act(async () => {
+        resolveSync!(true);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)